src/Rector/ClassMethod/RemoveUnusedRequestParamRector.php

```diff
@@ -113,4 +124,34 @@ public function refactor(Node $node): ?Node
 
         return null;
     }
+
+    private function isOverwritingAbstractFunction(ClassMethod $classMethod): bool
+    {
+        // Check if the method is overwriting an abstract method
+        $node = $classMethod->getAttribute(AttributeKey::PARENT_NODE);
+        if (! $node instanceof Class_) {
+            return false;
+        }
+        $methodName = $this->getName($classMethod);
+
+        $classReflection = $this->reflectionResolver->resolveClassReflection($classMethod);
+        if (! $classReflection instanceof ClassReflection) {
+            return false;
+        }
+        $reflectionParentClass = $classReflection->getParentClass();
+        if (! $reflectionParentClass instanceof ClassReflection) {
+            return false;
+        }
+        try {
+            $extendedMethodReflection = $reflectionParentClass->getNativeMethod($methodName);
+            assert($extendedMethodReflection instanceof PhpMethodReflection);
+            if (! $extendedMethodReflection->isAbstract()) {
```

Comment:
There is already service to verify it, see

https://github.com/rectorphp/rector-src/blob/5c1be93b80c3f7bfd8bd158ba0a55501a7acaefc/packages/FamilyTree/NodeAnalyzer/ClassChildAnalyzer.php#L52

Reply:
Nice find, i thought there must have been something but i couldn't find it i keep my private function to do the reflection and checking see `hasAbstractParentClassMethod` is that okay?

Reply:
That's fine as you need to get `ClassReflection` instance 👍